sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/integrationtest/BeamSqlDateFunctionsIntegrationTest.java

```diff
@@ -55,11 +56,13 @@ public Void apply(Iterable<Row> input) {
       Row row = iter.next();
       // LOCALTIME
       DateTime date = DateTime.now();
-      assertTrue(date.getMillis() - row.getDateTime(0).getMillis() < 1000);
-      assertTrue(date.getMillis() - row.getDateTime(1).getMillis() < 1000);
-      assertTrue(date.getMillis() - row.getDateTime(2).getMillis() < 1000);
-      assertTrue(date.getMillis() - row.getDateTime(3).getMillis() < 1000);
-      assertTrue(date.getMillis() - row.getDateTime(4).getMillis() < 1000);
+      long millis = date.getMillis();
+      int timeMillis = (int) (date.getMillis() % MILLIS_PER_DAY);
+      assertTrue(timeMillis - row.getDateTime(0).getMillis() < 1000);
+      assertTrue(millis - row.getDateTime(1).getMillis() < 1000);
+      assertTrue(millis - row.getDateTime(2).getMillis() < (MILLIS_PER_DAY + 1000));
```

Comment:
nit: can you add few comments here, it's hard to parse these assertions

Reply:
This is a bad test. I've cleaned it up significantly.